www/docs/getting_started.md

```diff
@@ -2,6 +2,151 @@
+
+Finally, if you'd like to just get your hands dirty and dive into the code, you can explore [examples](TODO: docusaurus link) of reactive services, [API docs](TODO: docusaurus link), or a before/after [example](TODO: docusaurus link) of retrofitting a legacy service with reactive features.
+
+## Tutorial
+
+This guide will walk you through getting your first reactive service up and running using the Skip runtime.
+It will also show how to write client code to read or subscribe to data from your reactive service.
+
+We aim to make this as beginner friendly as possible, but assume a basic understanding of imperative programming, including core JavaScript/Typescript syntax and semantics.
+
+### Reactive Programming
+
+Skip is a framework for building _reactive_ software which is responsive to changing inputs in real-time, efficient to execute, and intuitive to write and reason about.
+
+From the outside, reactive services can be treated similarly to streaming sources: they push real-time updates to clients and other subscribers, supporting smooth real-time user experiences.
+Unlike event-based streaming frameworks, though, Skip programs are written in a declarative style: instead of reasoning about updates directly, you write code that reasons about a "current" snapshot of state, and the Skip framework automatically processes changes and keeps everything in sync.
+
+In order to do so, Skip tracks dependencies in a _computation graph_ in which input changes can be propagated through to relevant outputs *without* re-evaluating any computation whose inputs haven't changed.
+
+Skip's core abstractions are best understood in terms of this computation graph.
+
+The primary data structure used in Skip is called a _collection_, and associates *keys* with one or more *values*.
+Collection keys and values are both immutable data stored in the Skip framework's native heap; in Typescript, any JSON-serializable value can be used. (i.e. primitives, arrays, objects, and nested combinations thereof)
+Collections can be accessed by key or manipulated to create new collections, which can either be _eager_ (updated in response to input changes) or _lazy_ (updated in response to queries for specific keys).
+
+These manipulations do _not_ mutate in-place but instead create a _new_ collection in the computation graph with a dependency edge from the input collection(s).
+For example, `collection.map(Foo)` creates a _new_ collection with the results of applying `Foo` to the data in `collection`, which is then updated in real time whenever the contents of `collection` change.
+(Along with `map`, Skip collections offer `mapReduce` and utilities like `merge`, `slice`, etc.)
+
+Maps and other operations over Skip collections can be sequenced and combined to create complex computation graphs which are maintained up-to-date by re-executing computation edges when inputs change.
```

Comment:
I don't think re-execute needs to be hyphenated, but can be. We should choose one and be consistent. Any strong opinions? I default to fewer hyphens.